### PR TITLE
Added support for Laravel 7 custom casts

### PR DESCRIPTION
**Pull request by belamov**
this pr adresses #904

the idea of improvement is simple:

now we get cast object as property's return type

we just check if this cast object implements laravel's `Illuminate\Contracts\Database\Eloquent\CastsAttributes` interface and if so then we return it's `get` method return type

just had one problem with this: if return type of `get` metod is not defined, but it has `@returns` tag, then returned type is not namespaced properly. i couldn't find a way to provide necessary context to `DocBlock` object. if anyone knows how to do it, that would be really cool to implement

### CODE REVIEW

File: src/Console/ModelsCommand.php [hunk 1]
```diff
@@ -866,4 +869,44 @@ private function getClassKeyword(ReflectionClass $reflection)
...
+
+        if (!$returnType->isBuiltin()) {
+            $type = '\\' . $type;
+        }
+
+        if ($returnType->allowsNull()) {
+            $type .= '|null';
+        }
+
+        return $type;
```
Comment (mfn) [inline src/Console/ModelsCommand.php:910]: I get the impression I've seen the code somewhere 😄

File: tests/Console/ModelsCommand/LaravelCustomCasts/DocblockReturnTypeTest.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+        $command = $this->app->make(ModelsCommand::class);
+
+        $tester = $this->runCommand($command, [
+            '--write' => true,
+        ]);
+
+        $this->assertSame(0, $tester->getStatusCode());
+        $this->assertEmpty($tester->getDisplay());
+//dd($actualContent);
```
Comment (mfn) [inline tests/Console/ModelsCommand/LaravelCustomCasts/DocblockReturnTypeTest.php:64]: debug stuff 💥

File: tests/Console/ModelsCommand/LaravelCustomCasts/ReturnTypeTest.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+class ReturnTypeTest extends AbstractModelsCommand
+{
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if (version_compare(Application::VERSION, '7.0', '<')) {
+            $this->markTestSkipped('This test requires Laravel 7.0 or higher');
+            return;
```
Comment (mfn) [inline tests/Console/ModelsCommand/LaravelCustomCasts/ReturnTypeTest.php:20]: Not needed, `markTestSkipped` works through exception bubbling

File: src/Console/ModelsCommand.php (hunk 1 above)
Comment (belamov) [reply src/Console/ModelsCommand.php:910]: yep, took it from your pr, but inly becouse it is not merged yet

File: tests/Console/ModelsCommand/LaravelCustomCasts/DocblockReturnTypeTest.php (hunk 2 above)
Comment (belamov) [reply tests/Console/ModelsCommand/LaravelCustomCasts/DocblockReturnTypeTest.php:64]: will fix

File: tests/Console/ModelsCommand/LaravelCustomCasts/ReturnTypeTest.php (hunk 3 above)
Comment (belamov) [reply tests/Console/ModelsCommand/LaravelCustomCasts/ReturnTypeTest.php:20]: found it only in `AbstractModelsCommand` but it checks for 6 version
```php
if (version_compare(Application::VERSION, '6.0', '<')) {
      $this->markTestSkipped('This test requires Laravel 6.0 or higher');
      return;
}
```

are you sure i can safely remove it? where is this exception debugging you are talking about?

Comment (mfn) [reply tests/Console/ModelsCommand/LaravelCustomCasts/ReturnTypeTest.php:20]: `markTestSkipped` throws a special exception recognized by PHPUnit, that's why a return (read: code afterwards) doesn't execute => https://github.com/sebastianbergmann/phpunit/blob/cb6b2d8cb5865b9576adcd9c595a8b4d005361b3/src/Framework/Assert.php#L2645-L2655

Comment (belamov) [reply tests/Console/ModelsCommand/LaravelCustomCasts/ReturnTypeTest.php:20]: yeah but to be honest, i cant see why i can drop this code
parent class check for 6 version, but my test should run only in 7

Comment (mfn) [reply tests/Console/ModelsCommand/LaravelCustomCasts/ReturnTypeTest.php:20]: It can/should be removed in the parent as well.

Comment (belamov) [reply tests/Console/ModelsCommand/LaravelCustomCasts/ReturnTypeTest.php:20]: sorry, but i don't follow why
travis runs test on multiple laravel versions, how else this test would be skipped without this code?

Comment (belamov) [reply tests/Console/ModelsCommand/LaravelCustomCasts/ReturnTypeTest.php:20]: or are you talking only about line 20? we can remove return statement, agreed

Comment (mfn) [reply tests/Console/ModelsCommand/LaravelCustomCasts/ReturnTypeTest.php:20]: Yes, exactly; apologies for being not more specific upfront!

**Comment by belamov** [comment]
refactored tests and added some additional checks in tests
thanks, @mfn for review!

**Comment by barryvdh** [comment]
Thanks! Unfortunately this has merge conflicts. Would it be possible to fix those, then I'll merge this.

**Comment by belamov** [comment]
done, replaced my code with @mfn's

**Comment by barryvdh** [comment]
Thanks!

**Pull request merged**